tests/v2/validation/rbac/globalrolesv2/globalroles_v2_test.go

```diff
@@ -0,0 +1,501 @@
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/rbac"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type GlobalRolesV2TestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	clusterCount int
+	user         *management.User
+	grbName      string
+	grName       string
+}
+
+func (gr *GlobalRolesV2TestSuite) TearDownSuite() {
+	gr.session.Cleanup()
+}
+
+func (gr *GlobalRolesV2TestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	gr.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(gr.T(), err)
+
+	gr.client = client
+}
+
+// Private helper function to create a Global Role with inheritedClusterRole and validate if GRBs, CRTBs, CRBs and RBs are created successfully.
+func (gr *GlobalRolesV2TestSuite) createAndTestGlobalRole(inheritedRoles []string) {
+	log.Info("Create a global role with inheritedClusterRoles.")
+	globalRole.Name = namegen.AppendRandomString("testgr")
+	globalRole.InheritedClusterRoles = inheritedRoles
+	createdGlobalRole, err := rbac.CreateGlobalRole(gr.client, localcluster, &globalRole)
+	require.NoError(gr.T(), err)
+	gr.grName = createdGlobalRole.Name
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(gr.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(gr.T(), err)
+	gr.user = createdUser
+
+	log.Info("Verify that the global role binding is created for the user.")
+	grbOwner, err := getGlobalRoleBindingForUser(gr.client, createdUser.ID)
+	require.NoError(gr.T(), err)
+	require.NotEmpty(gr.T(), grbOwner, "Global Role Binding not found for the user")
+	gr.grbName = grbOwner
+
+	log.Info("Verify that the cluster role template bindings are created for the downstream clusters.")
+	clusterNames, err := clusters.ListDownstreamClusters(gr.client)
+	require.NoError(gr.T(), err)
+	gr.clusterCount = len(clusterNames)
+	expectedCrtbCount := gr.clusterCount * len(inheritedRoles)
+	crtbs, err := listClusterRoleTemplateBindingsForGlobalRoleBinding(gr.client, grbOwner, expectedCrtbCount)
+	require.NoError(gr.T(), err)
+	actualCrtbCount := len(crtbs.Items)
+	require.Equal(gr.T(), expectedCrtbCount, actualCrtbCount, "Unexpected number of ClusterRoleTemplateBindings: Expected %d, Actual %d", expectedCrtbCount, actualCrtbCount)
+
+	log.Info("Verify that the cluster role bindings are created for the downstream cluster.")
+	expectedCrbCount := expectedCrtbCount
```

Comment:
I'm not sure if this is apt. For example, I created a global role with the inheritedClusterRoles:
  - cluster-member
  - projects-view

I noticed that there were 2 crtbs created but 3 crbs created in the downstream cluster namely:

- crb-cuapmrktvo                                         ClusterRole/cluster-member                                         
 - crb-pbpiprkini                                         ClusterRole/view                                                   
- crb-tg3w66tern                                         ClusterRole/projects-view                                          


First CRB has the CRTB for cluster-member and the other 2 CRBs have the CRTBs referring to the same. In this scenario the test would fail. Verifying expectedCRB and expectedRB might be a tad difficult as they differ for individual roles and won't be the same always..

Reply:
Based on our discussion, we will leave this code for CRBs and RBs as is.